tools/ports/libpng.py

```diff
@@ -34,13 +38,17 @@ def create(final):
     Path(dest_path, 'pnglibconf.h').write_text(pnglibconf_h)
     ports.install_headers(dest_path)
 
-    ports.build_port(dest_path, final, flags=['-sUSE_ZLIB=1'], exclude_files=['pngtest'], exclude_dirs=['scripts', 'contrib'])
+    flags = ['-sUSE_ZLIB=1']
+    if settings.USE_PTHREADS:
+      flags += ['-sSHARED_MEMORY=1']
```

Comment:
Why not USE_PTHREADS like the other ports do?

Reply:
I was following the suggestion of @kleisauke 

https://github.com/emscripten-core/emscripten/issues/16681#issuecomment-1096454239

libpng doesn't use pthreads, it just needs to be built with shared memory.

Reply:
Sure, but I think that is probably true for most of the ports that have `-mt` builds.   I don't know of any of them that actually start threads.. they just use atomics and pthread locks to make themselves thread safe.

I don't feel strongly but I think it would be better to be consistent across ports on this.